src/scheduler/agent/database.c

```diff
@@ -1042,6 +1042,39 @@ void database_job_priority(scheduler_t* scheduler, job_t* job, int priority)
   g_free(sql);
 }
 
+/**
+ * @brief Find s-nail version to check if mta is supported 
+ *
+ * \return 1 if mta is supported, 0 if not
+ */
+int check_mta_support()
+{
+  char cmd[] = "dpkg -s s-nail | grep -i 'Version' | awk '{print$2}' | cut -c -4";
+  char version_str[128];
+  char buf[128];
+  float version_float;
+  FILE *fp;
+
+  fp = popen(cmd, "r");
+  if(!fp)
+  {
+    WARNING("Unable to run the command '%s'.\n", cmd);
+    return 1;
+  }
+  while(fgets(buf, sizeof(buf), fp) != NULL)
+  {
+    strcpy(version_str,buf);
+  }
+  pclose(fp);
+  sscanf(version_str, "%f", &version_float);
+
+  if(version_float - 14.8 > 0.0001)
+  {
+    return 0;
+  }
+  return 1;
```

Comment:
Looks like the return values are swapped. Does not work on Xenial and Stretch until I swap the values.

Reply:
My bad 😅. Updated the code now

Reply:
Anyway, this is an interesting approach `version_float - 14.8 > 0.0001`